From 0254fcff6ab23e7be8924c02e23890529aa287b6 Mon Sep 17 00:00:00 2001
From: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
Date: Mon, 13 Apr 2026 11:57:59 -0300
Subject: [PATCH] ceph-csi-cephfs: update storageclass

This patch updates some fields and values for the cephfs storage class.

Signed-off-by: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
---
 .../templates/storageclass.yaml               | 114 ++++++++++--------
 1 file changed, 61 insertions(+), 53 deletions(-)

diff --git a/charts/ceph-csi-cephfs/templates/storageclass.yaml b/charts/ceph-csi-cephfs/templates/storageclass.yaml
index f791220..49ccac3 100644
--- a/charts/ceph-csi-cephfs/templates/storageclass.yaml
+++ b/charts/ceph-csi-cephfs/templates/storageclass.yaml
@@ -1,73 +1,81 @@
 {{- if .Values.storageClass.create -}}
+{{- $cephCsiDefault := .Values.storageClass }}
+{{- $defaults := .Values.classdefaults }}
+{{- range $sc := .Values.storageClasses }}
 apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
-  name: {{ .Values.storageClass.name }}
-{{- if .Values.storageClass.annotations }}
+  name: {{ $sc.name }}
+{{- if $cephCsiDefault.annotations }}
   annotations:
-{{ toYaml .Values.storageClass.annotations | indent 4 }}
+{{ toYaml $cephCsiDefault.annotations | indent 4 }}
 {{- end }}
   labels:
-    app: {{ include "ceph-csi-cephfs.name" . }}
-    chart: {{ include "ceph-csi-cephfs.chart" . }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
-    {{- with .Values.commonLabels }}{{ toYaml . | trim | nindent 4 }}{{- end }}
-provisioner: {{ .Values.driverName }}
+    app: {{ include "ceph-csi-cephfs.name" $ }}
+    chart: {{ include "ceph-csi-cephfs.chart" $ }}
+    release: {{ $.Release.Name }}
+    heritage: {{ $.Release.Service }}
+    {{- with $.Values.commonLabels }}{{ toYaml . | trim | nindent 4 }}{{- end }}
+provisioner: {{ $.Values.driverName }}
 parameters:
-  clusterID: {{ .Values.storageClass.clusterID }}
-  fsName: {{ .Values.storageClass.fsName }}
-{{- if .Values.storageClass.pool }}
-  pool: {{ .Values.storageClass.pool }}
+{{- with (first $.Values.csiConfig) }}
+  clusterID: {{ or $sc.clusterID .clusterID }}
+{{- end}}
+  fsName: {{ $sc.fs_name }}
+{{- if $sc.data_pool_name }}
+  pool:  {{ $sc.data_pool_name }}
 {{- end }}
-{{- if .Values.storageClass.encrypted }}
-  encrypted: "{{ .Values.storageClass.encrypted }}"
+{{- if $sc.encrypted }}
+  encrypted: "{{ $sc.encrypted }}"
 {{- end }}
-{{- if .Values.storageClass.encryptionKMSID }}
-  encryptionKMSID: {{ .Values.storageClass.encryptionKMSID }}
+{{- if $sc.encryptionKMSID }}
+  encryptionKMSID: {{ $sc.encryptionKMSID }}
 {{- end }}
-{{- if .Values.storageClass.fuseMountOptions }}
-  fuseMountOptions: "{{ .Values.storageClass.fuseMountOptions }}"
+{{- if $sc.fuseMountOptions }}
+  fuseMountOptions: "{{ $sc.fuseMountOptions }}"
+{{- else if $cephCsiDefault.fuseMountOptions }}
+  fuseMountOptions: "{{ $cephCsiDefault.fuseMountOptions }}"
 {{- end }}
-{{- if .Values.storageClass.kernelMountOptions }}
-  kernelMountOptions: "{{ .Values.storageClass.kernelMountOptions }}"
+{{- if $sc.kernelMountOptions }}
+  kernelMountOptions: "{{ $sc.kernelMountOptions }}"
+{{- else if $cephCsiDefault.kernelMountOptions }}
+  kernelMountOptions: "{{ $cephCsiDefault.kernelMountOptions }}"
 {{- end }}
-{{- if .Values.storageClass.mounter }}
-  mounter: "{{ .Values.storageClass.mounter }}"
+{{- if $sc.mounter }}
+  mounter: "{{ $sc.mounter }}"
+{{- else if $cephCsiDefault.mounter }}
+  mounter: "{{ $cephCsiDefault.mounter }}"
 {{- end }}
-{{- if .Values.storageClass.volumeNamePrefix }}
-  volumeNamePrefix: "{{ .Values.storageClass.volumeNamePrefix }}"
+{{- if $sc.volumeNamePrefix }}
+  volumeNamePrefix: "{{ $sc.volumeNamePrefix }}"
+{{- else if $cephCsiDefault.volumeNamePrefix }}
+  volumeNamePrefix: "{{ $cephCsiDefault.volumeNamePrefix }}"
 {{- end }}
-  csi.storage.k8s.io/provisioner-secret-name: {{ .Values.storageClass.provisionerSecret }}
-{{- if .Values.storageClass.provisionerSecretNamespace }}
-  csi.storage.k8s.io/provisioner-secret-namespace: {{ .Values.storageClass.provisionerSecretNamespace }}
-{{ else }}
-  csi.storage.k8s.io/provisioner-secret-namespace: {{ .Release.Namespace }}
-{{- end }}
-  csi.storage.k8s.io/controller-expand-secret-name: {{ .Values.storageClass.controllerExpandSecret }}
-{{- if .Values.storageClass.controllerExpandSecretNamespace }}
-  csi.storage.k8s.io/controller-expand-secret-namespace: {{ .Values.storageClass.controllerExpandSecretNamespace }}
-{{ else }}
-  csi.storage.k8s.io/controller-expand-secret-namespace: {{ .Release.Namespace }}
-{{- end }}
-  csi.storage.k8s.io/controller-publish-secret-name: {{ .Values.storageClass.controllerPublishSecret }}
-{{- if .Values.storageClass.controllerPublishSecretNamespace }}
-  csi.storage.k8s.io/controller-publish-secret-namespace: {{ .Values.storageClass.controllerPublishSecretNamespace }}
-{{ else }}
-  csi.storage.k8s.io/controller-publish-secret-namespace: {{ .Release.Namespace }}
-{{- end }}
-  csi.storage.k8s.io/node-stage-secret-name: {{ .Values.storageClass.nodeStageSecret }}
-{{- if .Values.storageClass.nodeStageSecretNamespace }}
-  csi.storage.k8s.io/node-stage-secret-namespace: {{ .Values.storageClass.nodeStageSecretNamespace }}
-{{ else }}
-  csi.storage.k8s.io/node-stage-secret-namespace: {{ .Release.Namespace }}
-{{- end }}
-reclaimPolicy: {{ .Values.storageClass.reclaimPolicy }}
-allowVolumeExpansion: {{ .Values.storageClass.allowVolumeExpansion }}
-{{- if .Values.storageClass.mountOptions }}
+  {{- $provisionerSecret := or $sc.provisionerSecret $defaults.adminSecretName }}
+  {{- $controllerExpandSecret := or $sc.controllerExpandSecret $defaults.adminSecretName }}
+  {{- $controllerPublishSecret := or $sc.controllerPublishSecret $controllerExpandSecret }}
+  {{- $nodeStageSecret := or $sc.nodeStageSecret $defaults.adminSecretName }}
+  csi.storage.k8s.io/provisioner-secret-name: {{ $provisionerSecret }}
+  csi.storage.k8s.io/provisioner-secret-namespace: {{ or $sc.provisionerSecretNamespace $.Release.Namespace }}
+  csi.storage.k8s.io/controller-expand-secret-name: {{ $controllerExpandSecret }}
+  csi.storage.k8s.io/controller-expand-secret-namespace: {{ or $sc.controllerExpandSecretNamespace $.Release.Namespace }}
+  csi.storage.k8s.io/controller-publish-secret-name: {{ $controllerPublishSecret }}
+  csi.storage.k8s.io/controller-publish-secret-namespace: {{ or $sc.controllerPublishSecretNamespace $.Release.Namespace }}
+  csi.storage.k8s.io/node-stage-secret-name: {{ $nodeStageSecret }}
+  csi.storage.k8s.io/node-stage-secret-namespace: {{ or $sc.nodeStageSecretNamespace $.Release.Namespace }}
+reclaimPolicy: {{ or $sc.reclaimPolicy $cephCsiDefault.reclaimPolicy }}
+allowVolumeExpansion: {{ or $sc.allowVolumeExpansion $cephCsiDefault.allowVolumeExpansion }}
+{{- if $sc.mountOptions }}
 mountOptions:
-  {{- range .Values.storageClass.mountOptions }}
+  {{- range $sc.mountOptions }}
   - {{ . }}
   {{- end }}
+{{- else if $cephCsiDefault.mountOptions }}
+mountOptions:
+  {{- range $cephCsiDefault.mountOptions }}
+  - {{ . }}
+  {{- end }}
+{{- end }}
+---
 {{- end }}
 {{- end -}}
--
2.34.1
